opta/core/kubernetes.py

```diff
@@ -387,12 +394,102 @@ def _aws_set_kube_config(layer: "Layer") -> None:
     return
 
 
-def get_cluster_name(layer: "Layer") -> Optional[str]:
-    outputs = get_terraform_outputs(layer)
-    cluster_name = outputs.get("parent.k8s_cluster_name") or outputs.get(
-        "k8s_cluster_name"
-    )
-    return cluster_name
+def get_cluster_name(layer: "Layer") -> str:
+    return f"opta-{layer.root().name}"
+
+
+def does_cluster_exist(layer: "Layer") -> bool:
+    if layer.is_stateless_mode() is True:
+        if logger.isEnabledFor(DEBUG):
+            logger.debug(
+                "does_cluster_exist called in stateless mode, verify implementation. See stack trace below:"
+            )
+            traceback.print_stack()
+
+    if layer.cloud == "aws":
+        return _aws_does_cluster_exist(layer)
+    elif layer.cloud == "google":
+        return _gcp_does_cluster_exist(layer)
+    elif layer.cloud == "azurerm":
+        return _azure_does_cluster_exist(layer)
+    elif layer.cloud == "local":
+        return _local_does_cluster_exist(layer)
+    else:
+        raise Exception(
+            f"Unknown cloud {layer.cloud}. Can not handle determining if cluster exists"
```

Comment:
to avoid this pattern, we can add an abstract method to CloudClient
https://github.com/run-x/opta/blob/main/opta/core/cloud_client.py#L8

it makes it clear about what needs to be implemented for each cloud.

but so also my other comment below, if we need this new cloud method or not.

Reply:
@RemyDeWolf took a lot of refactoring, but I think it's for the best-- take another look!